app/views/layouts/application.html.erb

```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title><%= yield :title %></title>
+    <title><%= full_title yield (:title) %></title>
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
```

Comment:
2 line này có tác dụng gì em ới

Reply:
Day la CSRF protection mac dinh khi tao project Rials, Em co doc qua nhung chua hieu ro a!

Reply:
tìm hiểu rồi note vào pull luôn nhé, pv có hỏi đấy

Reply:
vang a

Reply:
- CSRF (Cross-Site Request Forgery): Là một dạng tấn công cho phép một người dùng độc hại giả mạo các yêu cầu hợp pháp đến máy chủ dựa trên 1 phiên làm việc của người dùng
- CSRF có 2 thành phần chính: Thứ nhất, là một token duy nhất được nhúng vào trang HTML trong project, với token đó, nó sẽ được lưu trữ trong session cookie. 
- Khi người dùng tạo POST request lên hệ thống thì token ở trong html sẽ được gửi lên cùng với request đó. Rails sẽ nhận token được gửi lên cùng với request và token được lưu trong session cookie và so sánh để chắc chắn rằng chúng khớp với nhau.

-Tấn công SCRF -/cách hoạt động:
- Khi gửi một request HTTP, trình duyệt của nạn nhân sẽ nhận về Cookie. Các cookie thường được dùng để lưu trữ một session (Đọc thêm bài Session là gì? để hiểu thêm cách hoạt động của Session và cookies) để định danh người dùng không phải xác thực lại cho mỗi yêu cầu gửi lên.
- Nếu phiên làm việc đã xác thực của nạn nhân được lưu trữ trong một Cookie vẫn còn hiệu lực, và nếu ứng dụng không bảo mật dễ bị tấn công CSRF. Kẻ tấn công có thể thử dụng CSRF để chạy bất cứ requets nào với ứng dụng web mà ngay cả trang web không thể phân biệt được request nào là thực hay giả mạo.

Reply:
- CSP (Content Security Policy):  là 1 cách hữu hiệu để giảm thiểu hoặc loại bỏ hoàn toàn các lỗ hổng Cross Site Scripting (XSS). CSP có thể chặn inline script và các script từ những nguồn không đáng tin cậy. 
- Viec định nghĩa policy se thông qua 1 HTTP header chứa các rule dành cho tất cả các loại tài nguyên, các đoạn inline script se duoc tach ra các file riêng biệt.